inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/InstrumentationScopeResolver.java

```diff
@@ -0,0 +1,257 @@
+        }
+
+        return scopeMap;
+    }
+
+    /**
+     * Resolving the {@link InstrumentationScope} for the given parameters.
+     *
+     * @return The resolve {@link InstrumentationScope}.
+     */
+    private InstrumentationScope resolveScope(String key, InstrumentationScopeSettings scopeSettings) {
+        ElementMatcher.Junction<TypeDescription> typeMatcher = buildTypeMatcher(scopeSettings);
+        ElementMatcher.Junction<MethodDescription> methodMatcher = buildMethodMatcher(scopeSettings, typeMatcher);
+
+        if (typeMatcher == null) {
+            typeMatcher = any();
+            log.warn("|> The defined scope '{}' is not narrowing the type-scope, thus, matching ANY types! This may cause a negative performance impact!", key);
+        }
+
+        return new InstrumentationScope(typeMatcher, methodMatcher);
+    }
+
+    /**
+     * Creating an {@link ElementMatcher} which matches the specified methods.
+     */
+    private ElementMatcher.Junction<MethodDescription> buildMethodMatcher(InstrumentationScopeSettings scopeSettings, ElementMatcher.Junction<TypeDescription> typeMatcher) {
+        MatcherBuilder<MethodDescription> builder = new MatcherBuilder<>();
+
+        if (scopeSettings.getMethodScope() != null) {
+            scopeSettings.getMethodScope().forEach(m -> processMethod(builder, m));
+        }
+
+        if (builder.isEmpty()) {
+            builder.and(any());
+        }
+
+        ElementMatcher.Junction<MethodDescription> matcher = builder.build();
+
+        if (scopeSettings.getAdvanced() != null && scopeSettings.getAdvanced().getInstrumentOnlyInheritedMethods()) {
+            MatcherBuilder<TypeDescription> superBuilder = new MatcherBuilder<>();
+
+            if (scopeSettings.getTypeScope().getInterfaces() != null) {
+                scopeSettings.getTypeScope()
+                        .getInterfaces()
+                        .forEach(i -> superBuilder.or(buildNameMatcher(i)));
+            }
+            if (scopeSettings.getTypeScope().getSuperclass() != null) {
+                superBuilder.or(buildNameMatcher(scopeSettings.getTypeScope().getSuperclass()));
+            }
+
+            if (!superBuilder.isEmpty()) {
+                matcher = isOverriddenFrom(superBuilder.build()).and(builder.build());
+            }
+        }
+
+        return matcher;
+    }
+
+    /**
+     * Processing a single {@link MethodMatcherSettings} and adding it to the given {@link MatcherBuilder}.
+     */
+    private void processMethod(MatcherBuilder<MethodDescription> builder, MethodMatcherSettings matcherSettings) {
+        MatcherBuilder<MethodDescription> innerBuilder = new MatcherBuilder<>();
+
+        innerBuilder.and(matcherSettings.getIsConstructor(), isConstructor());
+        innerBuilder.and(buildVisibilityMatcher(matcherSettings.getVisibility()));
+        innerBuilder.and(buildArgumentMatcher(matcherSettings.getArguments()));
+
+        if (!matcherSettings.getIsConstructor()) {

```

Comment:
what happens if isConstructor is null? If you don't want to allow it add a @NotNull annotation to the config

Reply:
Changed to primitive type

Reply:
Changed it back and added the NotNull annotation. By this the user can write "is-constructor" in the configuration otherwise, if it's a primitive, the config-settings is just named "constructor" which is less expressive and another pattern than the other boolean values in that config element